Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -469,14 +469,22 @@ public struct ConvertAction: Action, RecreatingContext {
         }
 
         var didEncounterError = analysisProblems.containsErrors || conversionProblems.containsErrors
-        if try context.renderRootModules.isEmpty {
+        if (try context.renderRootModules + context.rootTechnologies).isEmpty {
+            // Check if the converted documentation is a tutorial to provide
+            // a specific error message.
+            let hasTutorial = context.knownPages.contains(where: {
+                guard let kind = try? context.entity(with: $0).kind else { return false }
+                return kind == .tutorial || kind == .tutorialArticle
+            })
             postConversionProblems.append(
                 Problem(
                     diagnostic: Diagnostic(
                         severity: .warning,
                         identifier: "org.swift.docc.MissingTechnologyRoot",
-                         summary: "No TechnologyRoot to organize article-only documentation.",
-                         explanation: "Article-only documentation needs a TechnologyRoot page (indicated by a `TechnologyRoot` directive within a `Metadata` directive) to define the root of the documentation hierarchy."
+                         summary: "There was no root found for this documentation catalog.",
```

Comment:
I don't know if fixits/solutions support defining new files or if they can only modify existing files. If it's possible to define a new file in a fixit/solution then maybe we could reuse the init command code to fill in most of the necessary information for the developer

```swift
let tableOfContentsFilename = "table-of-contents.tutorial"
let source = rootURL?.appendingPathComponent(tableOfContentsFilename)

Solution(
    summary: "Create a `@Tutorials` table of content page",
    replacements: [
        Replacement(
            range: .init(line: 1, column: 1, source: source) ..< .init(line: 1, column: 1, source: source),
            replacement: CatalogTemplateKind.tutorialTemplateFiles(converter.firstAvailableBundle()?.displayName ?? "Project Name")[tableOfContentsFilename]! // this file name is known to exist in the dictionary
        )
    ]
)
```

Reply:
I added your suggestion and tweaked how `formattedDiagnosticSource` works in order to be able to suggest addition of new content files without having to refactor the `Replacement` struct.

```swift
guard let diagnosticRange = diagnostic.range
      else {
          // If the replacement operation involves adding new files,
          // emit the file content as an addition instead of a replacement.
          //
          // Example:
          // --> /path/to/new/file.md
          // Summary
          // suggestion:
          // 0 + Addition file and
          // 1 + multiline file content.
          var addition = ""
          solutions.forEach { solution in
              addition.append("\n" + solution.summary)
              solution.replacements.forEach { replacement in
                  let solutionFragments = replacement.replacement.split(separator: "\n")
                  addition += "\nsuggestion:\n" + solutionFragments.enumerated().map {
                      "\($0.offset) + \($0.element)"
                  }.joined(separator: "\n")
              }
          }
          return "\n--> \(formattedSourcePath(url))\(addition)"
      }
```

Since the diagnostic does not belongs to an existing range in the source code the `diagnostic.range` will be nil and  we log the suggestion.

This allows us to use fixits/solutions for defining new files instead of only modify existing files.

The other possible improvement to do here is making `range` in `Replacement` optional, but I don't know if that might be a breaking API change. 

Implemented here https://github.com/apple/swift-docc/pull/812/commits/d7720d96b668f7df2ac407f334948fcceb1d4e69.